src/focus.service.ts

```diff
@@ -509,17 +509,16 @@ export class FocusService {
   private isFocusable(el: HTMLElement) {
     const role = el.getAttribute('role');
     const tabIndex = el.tabIndex;
-    let visible = el.offsetParent !== null;
-
-    for (let parent = el; parent !== this.root; parent = parent.parentElement) {
-      if (parent.classList.contains('hidden')) {
-        visible = false;
-        break;
-      }
-    }
+    const style = window.getComputedStyle(el);
+    let visible = style.display !== 'none' && style.visibility !== 'hidden';
 
     if (visible) {
-      visible = window.getComputedStyle(el).display !== 'none';
+      for (let parent = el; parent !== this.root; parent = parent.parentElement) {
+        if (window.getComputedStyle(parent).opacity === '0') {
+          visible = false;
+          break;
```

Comment:
You could just return false from inside here instead of needing visibility check + breaking